### PR TITLE
Update project dependencies

### PR DESCRIPTION
**Pull request by serious6**
* updates the used dependencies to the latest version
* fixes a small javadoc warning generated by `mvn site`

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@serious6__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by vbauer** [comment]
:+1:

File: ews-java-api.iml [hunk 1]
```diff
@@ -11,12 +11,28 @@
...
+    <orderEntry type="library" scope="TEST" name="Maven: org.slf4j:slf4j-api:1.7.12" level="project" />
+    <orderEntry type="library" scope="TEST" name="Maven: ch.qos.logback:logback-classic:1.1.3" level="project" />
+    <orderEntry type="library" scope="TEST" name="Maven: ch.qos.logback:logback-core:1.1.3" level="project" />
+    <orderEntry type="library" scope="TEST" name="Maven: org.slf4j:jcl-over-slf4j:1.7.12" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.httpcomponents:httpclient:4.4.1" level="project" />
+    <orderEntry type="library" name="Maven: commons-codec:commons-codec:1.9" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.httpcomponents:httpcore:4.4.1" level="project" />
+    <orderEntry type="library" name="Maven: commons-logging:commons-logging:1.2" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.commons:commons-lang3:3.4" level="project" />
+    <orderEntry type="library" name="Maven: joda-time:joda-time:2.8" level="project" />
```
Comment (vboctor) [inline ews-java-api.iml]: Are there duplicated entries?

**Comment by vboctor** [comment]
:+1:

File: ews-java-api.iml (hunk 1 above)
Comment (serious6) [inline ews-java-api.iml]: thanks for the reminder.

**Pull request merged**